src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentEventHandlerLoweringPass.cs

```diff
@@ -54,7 +54,7 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
                     continue;
                 }
 
-                if (node.TagHelper.IsEventHandlerTagHelper())
+                if (node.TagHelper.IsEventHandlerTagHelper() && node.IsDirectiveAttribute)
```

Comment:
Are these extra checks needed in practice? or are they just for completeness?

Reply:
I guess the reason why I'm asking is that I would expect to see a comment/explanation here if they are really required.

Reply:
No they aren't really needed in practice. They are just here to catch bugs in how we create descriptors. This will only really be useful if we ever make directive attributes extensible.